test/compiler/inline.jl

```diff
@@ -672,3 +672,8 @@ end
         success(pipeline(Cmd(cmd); stdout=stdout, stderr=stderr))
     end
 end
+
+# Issue #42264 - crash on certain union splits
+let f(x) = (x...,)
+    @test code_typed(f, Tuple{Union{Int64, CartesianIndex{1}, CartesianIndex{3}}})[1][2] == Tuple{Int64}
```

Comment:
For seemingly random types in tests, it's useful to annotate what specific thing we're relying on here. It helps catch outdated tests if underlying assumptions (e.g. that `iterate(::CartesianIndex)` throws) get invalidated.
```suggestion
    # Test splatting with a Union of non-{Tuple, SimpleVector} types that require creating new `iterate` calls
    # in inlining. For this particular case, we're relying on `iterate(::CaretesianIndex)` throwing an error, such
    # the the original apply call is not union-split, but the inserted `iterate` call is.
    @test code_typed(f, Tuple{Union{Int64, CartesianIndex{1}, CartesianIndex{3}}})[1][2] == Tuple{Int64}
```